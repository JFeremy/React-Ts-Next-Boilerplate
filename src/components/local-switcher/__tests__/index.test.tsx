import { render, screen } from '@testing-library/react';

import LocaleSwitcher from '..';

describe('LocaleSwitcher', () => {
  const expectedText = 'Lorem ipsum dolor sit amet.';
  it('shoud have expected text and correct link', async () => {
    render(<LocaleSwitcher title={expectedText} />); // ARRANGE

    const text = screen.getByText(expectedText); // ACT
    const links = screen.getAllByRole('link');

    expect(text).toBeInTheDocument(); // ASSERT
    expect(links).toHaveLength(2);
    expect(links[0]).toHaveProperty('href', 'http://localhost/en');
    expect(links[0]).toHaveProperty('text', 'en');
    expect(links[1]).toHaveProperty('href', 'http://localhost/fr');
    expect(links[1]).toHaveProperty('text', 'fr');
  });

  it('component matches snapshot', async () => {
    const view = render(<LocaleSwitcher title={expectedText} />); // ARRANGE
    expect(view.container).toMatchSnapshot();
  });
});
