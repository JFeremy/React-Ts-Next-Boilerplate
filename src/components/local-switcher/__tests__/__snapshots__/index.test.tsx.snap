// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSwitcher component matches snapshot 1`] = `
<div>
  <div
    class="font_anybody"
  >
    <p>
      Lorem ipsum dolor sit amet.
    </p>
    <ul>
      <li>
        <a
          href="/en"
        >
          en
        </a>
      </li>
      <li>
        <a
          href="/fr"
        >
          fr
        </a>
      </li>
    </ul>
  </div>
</div>
`;
